File: backend/test/src/main/java/com/accenture/test/Domain/Empresa/DTO/EmpresaFornResponseDTO.java
package com.accenture.test.Domain.Empresa.DTO;

import com.accenture.test.Domain.Fornecedor.DTO.FornecedorResponseDTO;

import java.util.List;
import java.util.UUID;

public record EmpresaFornResponseDTO(
        UUID id,
        String cnpj,
        String nome_fantasia,
        String cep,
        List<FornecedorResponseDTO> fornecedores
) {}

File: frontend/app/src/interfaces/Empresa/EmpresaFornResponseDTO.ts
import { FornecedorResponseDTO } from "../Fornecedor/FornecedorResponseDTO";

export interface EmpresaFornResponseDTO {
   id: string;
   cnpj: string;
   nome_fantasia: string;
   cep: string;
   fornecedores: FornecedorResponseDTO[]
}

export interface EmpresaPagFornResponseDTO {
   dados: EmpresaFornResponseDTO[],
   paginaAtual: number,
   totalPaginas: number,
   paginasRestantes: number
}
